pkg/controllers/validatingadmissionpolicy-generate/controller.go

```diff
@@ -461,6 +463,11 @@ func (c *controller) reconcile(ctx context.Context, logger logr.Logger, key, nam
 			c.updateClusterPolicyStatus(ctx, *policy, false, err.Error())
 			return err
 		}
+		if observedVAP.Status.TypeChecking.ExpressionWarnings != nil {
```

Comment:
We shouldn't use the type-checking warnings from VAPs. We need to block the creation of Kyverno policy if these warnings exist. This means that we have to do the type-checking itself.

Reply:
on it ;)